src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -597,12 +597,12 @@ private BoundExpression TransformCompoundAssignmentLHS(BoundExpression originalL
             return variableTemp;
         }
 
-        private BoundExpression BoxReceiver(BoundExpression rewrittenReceiver, NamedTypeSymbol memberContainingType)
+        private BoundExpression BoxTypeParameterReceiver(BoundExpression rewrittenReceiver, NamedTypeSymbol memberContainingType)
         {
             return MakeConversionNode(
                 rewrittenReceiver.Syntax,
                 rewrittenReceiver,
-                Conversion.Boxing,
+                Conversion.TypeParameterBoxing,
```

Comment:
Sure, i can do that.  But the method's name is "BoxTypeParameterReceiver".  Seems really bad if it didn't get a type parameter.  But i suppose that's why the assert is good :)